keel-api/src/test/resources/examples/cluster-example.yml

```diff
@@ -14,6 +14,7 @@ spec:
       type: DEB
   locations:
     accountName: test
+    vpcName: vpc0
```

Comment:
I ended up adding this to cluster for consistency. Not sure if that's desirable

Reply:
Can this replace the need for `subnet` in the regions map?

Reply:
No, I think a VPC has many subnets as I understand it

Reply:
A VPC can have many subnets but I don't think that prevents us from attempting to provide an opinionated default. In the absence of either a subnet or vpc, we could attempt to default to a subnet of "internal (vpc0)" which will correlate to a vpcId if it exists. Given just a `vpcName`, we could attempt to find a subnet name matching `"internal ($vpcName)"`.

I think it makes more sense not to require `vpcName` or `subnet`, with both derived by default from a subnet with the purpose of `internal (vpc0)` if present. If you need to deploy elsewhere, only specifying a different subnet should always be sufficient. There are subnets without a listed purpose but I don't think those are properly supported by clouddriver anyways, they need to be tagged first.

Reply:
That all makes sense. I guess my only question is whether that reflects a Netflix-only opinion of network configuration. Can/should we assume that's also valid for OSS users? I guess we could make those defaults configurable down the road.

Reply:
I think Spinnaker is opinionated enough that a "purpose" is required for use in OSS, but not about its contents. What do you think of Spring configuration properties for potential defaults like this that aren't easily resolved by "all the things available", like AZ's?

Reply:
Yes, that's where I was heading. Although, I feel like I'd like to do that as a separate piece of work.

Reply:
BTW, I couldn't find any information in the OSS docs about how VPCs and subnets are supposed to be setup, what tags are expected to exist on those resources for Spinnaker to find them (including the naming conventions above), etc. The closest thing I found was [this](https://www.spinnaker.io/setup/install/providers/aws/aws-ec2/) but it only discusses setting up the accounts.

Reply:
For future reference (thanks @asher!): https://docs.armory.io/spinnaker-install-admin-guides/aws-subnets/#configuring-subnets